cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -384,13 +384,13 @@ public long getApplicationCount() throws IOException {
 
   @Nullable
   public ApplicationMeta getLatest(ApplicationReference appReference) throws IOException {
-    Range range = getNamespaceAndApplicationRange(appReference.getNamespace(), appReference.getApplication());
     // scan based on: latest field set to true
-    Field<?> indexField = Fields.stringField(StoreDefinition.AppMetadataStore.LATEST_FIELD, "true");
+    Range range = getLatestApplicationRange(appReference);
     StructuredTable appSpecTable = getApplicationSpecificationTable();
 
+    // Get the most recently created latest version
     try (CloseableIterator<StructuredRow> iterator =
-           appSpecTable.scan(range, 1, Collections.singletonList(indexField))) {
+           appSpecTable.scan(range, 1, StoreDefinition.AppMetadataStore.CREATION_TIME_FIELD, SortOrder.DESC)) {
```

Comment:
This scan method has the same effect as before. Just adding another layer of guarantee that we also want the most recently deployed "latest" version

Reply:
umm I don't think we can even optimize the number of times we call getLatest in the deployment process - for ex: In `createApplicationVersion` we need to call getLatest again (since needs to be within the transaction)

Reply:
this should be the same behavior, but it is still safer and more understandable to return whatever has latest = true. In case there is some bug where the latest column is not set for the greatest creation time, returning the row with latest=true will be more correct, as that is what gets returned in the get /apps list, and that is the row that we would want to change to latest=false when a new version is added.

Reply:
Yes this scan is returning `whatever has latest = true`, check line 399 (the scan range has both keys and non-key field `latest`). The change is on top of that, we're also DESC sorting all the `latest==true` versions, return the one with greatest creation_time.

1. When in happy situation, there's only one true latest version. In case there's something wrong like someone outside CDAP manually inserted a second latest=true version into DB, we should return that new one

2. `In case there is some bug where the latest column is not set for the greatest creation time, returning the row with latest=true will be more correct` -- this is still valid after the change

Reply:
I see, didn't realize the Range included latest=true.

Reply:
@albertshau Thanks for the review and approval!